packages/imperative/src/plugins/PluginManagementFacility.ts

```diff
@@ -257,89 +258,87 @@ export class PluginManagementFacility {
-                    const overrideErrMsg = `You attempted to override the "${setting}" setting ` +
-                        `with a plugin named "${pluginName}" that is not installed and loadable.` +
-                        `\nWe will use a "${setting}" that purposely fails until you reconfigure.`;
-                    Logger.getImperativeLogger().error(overrideErrMsg);
-
-                    /* We need to assign a class (which always shows an error when
-                     * the CLI tries to use credentials) into the current override setting.
-                     * We also need to embed our error message into that class. We cannot
-                     * create a new object from a class and pass the error into its
-                     * constructor, because the CredentialManagerFactory takes a class and
-                     * it calls the constructor of our supplied class. Thus we need an
-                     * anonymous class so that we can access our 'overrideErrMsg' variable.
-                     * Our trick is that we simply throw an error in the constructor
-                     * of our anonymous class. The CredentialManagerFactory catches
-                     * our error, and places it into its InvalidCredentialManager,
-                     * which in turn shows our error every time the CLI tries to use
-                     * credentials. Finally since lint complains about more than one
-                     * class in a file, we have to temporarily turn off that lint error.
+
+                if (settingNm === CredentialManagerOverride.CRED_MGR_SETTING_NAME) {
+                    /* For credMgr override, the value of the setting is the override display name.
+                     * We must use the display name to find this override within our loadedOverrides.
+                     * We must find the plugin name from within our known credMgr overrides.
                      */
-                    (this.mPluginOverrides as any)[setting] = class {
-                        constructor() {
-                            throw overrideErrMsg;
-                        }
-                    };
+                    const credMgrInfo = CredentialManagerOverride.getCredMgrInfoByDisplayName(overrideDispNm);
+                    if ( credMgrInfo === null) {
+                        // The credMgr override name in settings/.imperative.json is unknown
+                        this.useOverrideThatFails(settingNm, overrideDispNm, overridePluginNm,
+                            `The "${settingNm}" named "${overrideDispNm}" is not a known credential manager override.`
+                        );
+                        continue;
+                    }
+
+                    // record the known plugin name that we found for this display name
+                    overridePluginNm = credMgrInfo.credMgrPluginName;
+                }
+
+                if (!Object.prototype.hasOwnProperty.call(loadedOverrides, overridePluginNm)) {
```

Comment:
There are 24 such pre-existing calls in this file. I tend not to change the style being used throughout a file for one new addition. We could treat such a wholesale change as technical debt to be done within this PR, and confirm whether the full set of automated test suites are successful.

Reply:
Sorry for the misleading suggestion! Turns out that `Object.hasOwn` was introduced in Node 16, and it will not be supported with Node 14. 😔 

The `in` operator is another alternative, but there's no sense in expanding the scope of this PR. Thanks for taking a look into it.

Reply:
Object.hasOwn is not available in NodeJS 14. We must postpone any such changes until we no longer support NodeJS 14.